connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSinkTask.java

```diff
@@ -557,7 +557,7 @@ private void deliverMessages() {
             // Let this exit normally, the batch will be reprocessed on the next loop.
         } catch (Throwable t) {
             log.error("{} Task threw an uncaught and unrecoverable exception. Task is being killed and will not "
-                    + "recover until manually restarted.", this, t);
+                    + "recover until manually restarted. {}", this, t.getMessage());
```

Comment:
Thanks for the suggested fix, @NarekDW. 

The convention for the various Log4J method calls is that when the exception is passed as the _(n+1)_ parameter where _n_ is the number of `{}` placeholders in the message string, then the exception's stack trace is included in the log immediately following the error message.

In this particular case, the first `{}` placeholder is replaced with the result of `this.toString()`, and before this change there is no second `{}` placeholder and so the `error(...)` method call will generate a stack trace with the exception `t`.

You are correct in that the error returned by the Connect API does not include the message of the original exception. That probably is helpful, but your proposed change would cause Log4J to not log the stack trace. I think the following change would give us both and modify the message to help to identify the problem:
```suggestion
                    + "recover until manually restarted. Error: {}", this, t.getMessage(), t);
```

Reply:
@rhauch good point, thank you! I missed that the last parameter generates stack trace.
Added suggested change.